packages/context/src/value-promise.ts

```diff
@@ -218,3 +217,42 @@ export function tryWithFinally<T>(
   }
   return result;
 }
+
+/**
+ * Resolve an iterator of source values into a result until the evaluator
+ * returns `true`
+ * @param source The iterator of source values
+ * @param resolver The resolve function that maps the source value to a result
+ * @param evaluator The evaluate function that decides when to stop
+ */
+export function resolveUntil<T, V>(
+  source: Iterator<T>,
+  resolver: (sourceVal: T) => ValueOrPromise<V | undefined>,
+  evaluator: (sourceVal: T, targetVal: V | undefined) => boolean,
+): ValueOrPromise<V | undefined> {
+  const next = source.next();
+  if (next.done) return undefined; // End of the iterator
+  const sourceVal = next.value;
+  const valueOrPromise = resolver(sourceVal);
+  return resolveValueOrPromise(valueOrPromise, v => {
+    if (evaluator(sourceVal, v)) return v;
+    else return resolveUntil(source, resolver, evaluator);
+  });
+}
+
+/**
+ * Resolve a value or promise with a function that produces a new value or
+ * promise
+ * @param valueOrPromise The value or promise
+ * @param resolver A function that maps the source value to a value or promise
+ */
+export function resolveValueOrPromise<T, V>(
```

Comment:
I find the name `resolveValueOrPromise` confusing. In my mind, we _resolve_ values from a context (IoC container), but there is no context involved here. I am proposed to rename this function to `mapValueOrPromise` or perhaps `transformValueOrPromise`.

OTOH, it is true that Promises can be resolved too, so maybe "resolve" is no that bad.

@strongloop/sq-lb-apex Thoughts?

Reply:
I renamed it to `transformValueOrPromise`